provider/infoblox/infoblox.go

```diff
@@ -140,8 +148,14 @@ func NewInfobloxProvider(ibStartupCfg StartupConfig) (*ProviderConfig, error) {
 	}
 
 	authCfg := ibclient.AuthConfig{
-		Username: ibStartupCfg.Username,
-		Password: ibStartupCfg.Password,
+		Username:   ibStartupCfg.Username,
+		Password:   ibStartupCfg.Password,
+		ClientCert: nil,
+		ClientKey:  nil,
+	}
+
+	if authCfg.Username != "" {
```

Comment:
Just thought of it: `Username` may be not empty while `Password `is. This will result into an attempt of the password based authentication while password is not there. I don't know how Infoblox golang client will react to this. Maybe it would make sense to check whether both username and password are set or not set both.